File: SYP.Web/Modules/Order/Order/Editor/OrderEditorColumns.cs
namespace SYP.Order.Columns;

[ColumnsScript("Order.OrderEditor")]
[BasedOnRow(typeof(OrderRow), CheckNames = true)]
public class OrderEditorColumns
{
    [EditLink, DisplayName("Db.Shared.RecordId"), AlignRight]
    public int Id { get; set; }
    [EditLink]
    public string OrderNumber { get; set; }
    public string CustomerCode { get; set; }
    public string ManagerUserUsername { get; set; }
    public int Status { get; set; }
    public DateTime OrderDate { get; set; }
    public decimal TotalAmount { get; set; }
    public decimal DiscountPercentage { get; set; }
    public decimal DiscountAmount { get; set; }
    public decimal NetAmount { get; set; }
    public string CurrencyCode { get; set; }
    public string Notes { get; set; }
    public string RejectReason { get; set; }
    public DateTime InsertDate { get; set; }
    public int InsertUserId { get; set; }
    public DateTime UpdateDate { get; set; }
    public int UpdateUserId { get; set; }
}

File: SYP.Web/Modules/ServerTypes/Order/OrderEditorColumns.ts
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { OrderRow } from "./OrderRow";
import { OrderStatus } from "./OrderStatus";

export interface OrderEditorColumns {
    Id: Column<OrderRow>;
    OrderNumber: Column<OrderRow>;
    CustomerCode: Column<OrderRow>;
    Status: Column<OrderRow>;
    OrderDate: Column<OrderRow>;
    TotalAmount: Column<OrderRow>;
    DiscountPercentage: Column<OrderRow>;
    DiscountAmount: Column<OrderRow>;
    NetAmount: Column<OrderRow>;
    CurrencyCode: Column<OrderRow>;
    Notes: Column<OrderRow>;
    RejectReason: Column<OrderRow>;
    InsertDate: Column<OrderRow>;
    InsertUserId: Column<OrderRow>;
    UpdateDate: Column<OrderRow>;
    UpdateUserId: Column<OrderRow>;
}

export class OrderEditorColumns extends ColumnsBase<OrderRow> {
    static readonly columnsKey = 'Order.OrderEditor';
    static readonly Fields = fieldsProxy<OrderEditorColumns>();
}

[OrderStatus]; // referenced types